presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -565,7 +565,7 @@ public PlanWithProperties visitDistinctLimit(DistinctLimitNode node, PreferredPr
                         gatheringExchange(
                                 idAllocator.getNextId(),
                                 REMOTE_STREAMING,
-                                new DistinctLimitNode(child.getNode().getSourceLocation(), idAllocator.getNextId(), child.getNode(), node.getLimit(), true, node.getDistinctVariables(), node.getHashVariable(), 0)),
+                                new DistinctLimitNode(child.getNode().getSourceLocation(), idAllocator.getNextId(), child.getNode(), node.getLimit(), true, node.getDistinctVariables(), node.getHashVariable(), node.getTimeoutMillis())),
```

Comment:
I know this was done in previous commit, but I don't think PlanNode should be storing timeout information. We should look for alternate mechanism to pass this information. I would suggest to add a TODO to cleanup this.

Reply:
Yeah I thought about it - it's a pain to deal with all the middle-tier stuff (from Feilong's experience).

Reply:
We need a better high level API that has these limits at query level (and not error out instead return partial results). That will help a lot of UIs built on Presto. It's a long term vision.

Reply:
We should be able to pass this information from other means. If you are not cleaning this in this PR, can you please add a TODO?

Reply:
SHould be - but not right npw

Reply:
+1 it's strange to have a timeout in the plan node.  one possibly better option to have some kind of "best effort, it's okay if we don't get the full results" boolean for the plan node and the operator figures out what to do based on that.

Reply:
Done added in the optimization rule itself as a comment as that's the only use case for now.